File: k.yyup.com/client/src/scripts/fix-parent-pages.ts
/**
 * 修复家长端页面脚本
 * 为所有页面添加UnifiedCenterLayout并替换硬编码值
 */

import * as fs from 'fs';
import * as path from 'path';

const parentCenterPath = path.join(__dirname, '../pages/parent-center');

// 需要修复的页面列表
const pagesToFix = [
  'activities/index.vue',
  'ai-assistant/index.vue',
  'assessment/Academic.vue',
  'assessment/Doing.vue',
  'assessment/GrowthTrajectory.vue',
  'assessment/Report.vue',
  'assessment/SchoolReadiness.vue',
  'assessment/Start.vue',
  'assessment/components/GameComponent.vue',
  'assessment/games/AttentionGame.vue',
  'assessment/games/LogicGame.vue',
  'assessment/games/MemoryGame.vue',
  'assessment/index.vue',
  'children/FollowUp.vue',
  'children/Growth.vue',
  'children/index.vue',
  'communication/smart-hub.vue',
  'feedback/ParentFeedback.vue',
  'games/achievements.vue',
  'games/components/GameCard.vue',
  'games/index.vue',
  'games/play/AnimalObserver.vue',
  'games/play/ColorSorting.vue',
  'games/play/DinosaurMemory.vue',
  'games/play/DollhouseTidy.vue',
  'games/play/FruitSequence.vue',
  'games/play/PrincessGarden.vue',
  'games/play/PrincessMemory.vue',
  'games/play/RobotFactory.vue',
  'games/play/SpaceTreasure.vue',
  'games/records.vue',
  'profile/index.vue',
  'share-stats/index.vue'
];

console.log(`🔧 开始修复 ${pagesToFix.length} 个页面...\n`);

let fixedCount = 0;
let errorCount = 0;

pagesToFix.forEach((page, index) => {
  const filePath = path.join(parentCenterPath, page);
  
  try {
    if (!fs.existsSync(filePath)) {
      console.log(`⚠️ ${index + 1}. ${page} - 文件不存在`);
      errorCount++;
      return;
    }

    let content = fs.readFileSync(filePath, 'utf-8');
    const originalContent = content;

    // 检查是否已经使用了UnifiedCenterLayout
    if (content.includes('UnifiedCenterLayout')) {
      console.log(`✅ ${index + 1}. ${page} - 已使用UnifiedCenterLayout`);
      fixedCount++;
      return;
    }

    // 替换硬编码的颜色值
    content = content.replace(/#[0-9a-fA-F]{6}/g, (match) => {
      // 根据颜色值映射到设计令牌
      const colorMap: { [key: string]: string } = {
        '#FF0000': 'var(--color-danger-500)',
        '#00FF00': 'var(--color-success-500)',
        '#0000FF': 'var(--color-primary-500)',
        '#FFFF00': 'var(--color-warning-500)',
        '#FF6B6B': 'var(--color-danger-300)',
        '#51CF66': 'var(--color-success-300)',
        '#4ECDC4': 'var(--color-info-300)',
      };
      return colorMap[match.toUpperCase()] || match;
    });

    // 替换硬编码的rgb颜色值
    content = content.replace(/rgb\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)/g, 
      'var(--color-primary-500)');

    // 替换硬编码的px值为设计令牌
    content = content.replace(/(\d+)px/g, (match, num) => {
      const px = parseInt(num);
      if (px === 4) return 'var(--spacing-xs)';
      if (px === 8) return 'var(--spacing-sm)';
      if (px === 12) return 'var(--spacing-md)';
      if (px === 16) return 'var(--spacing-lg)';
      if (px === 20) return 'var(--spacing-xl)';
      if (px === 24) return 'var(--spacing-2xl)';
      if (px === 32) return 'var(--spacing-3xl)';
      return match;
    });

    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf-8');
      console.log(`✅ ${index + 1}. ${page} - 已修复`);
      fixedCount++;
    } else {
      console.log(`⚠️ ${index + 1}. ${page} - 无需修复`);
    }
  } catch (error) {
    console.log(`❌ ${index + 1}. ${page} - 错误: ${error}`);
    errorCount++;
  }
});

console.log(`\n📊 修复完成:`);
console.log(`  ✅ 成功: ${fixedCount}`);
console.log(`  ❌ 失败: ${errorCount}`);
console.log(`  📝 总计: ${pagesToFix.length}`);

File: k.yyup.com/fix_parent_pages.py
#!/usr/bin/env python3

import os
import re

parent_center_path = '/home/zhgue/kyyupgame/k.yyup.com/client/src/pages/parent-center'

pages_to_fix = [
    'activities/index.vue',
    'ai-assistant/index.vue',
    'assessment/Academic.vue',
    'assessment/Doing.vue',
    'assessment/GrowthTrajectory.vue',
    'assessment/Report.vue',
    'assessment/SchoolReadiness.vue',
    'assessment/Start.vue',
    'assessment/components/GameComponent.vue',
    'assessment/games/AttentionGame.vue',
    'assessment/games/LogicGame.vue',
    'assessment/games/MemoryGame.vue',
    'assessment/index.vue',
    'children/FollowUp.vue',
    'children/Growth.vue',
    'children/index.vue',
    'communication/smart-hub.vue',
    'feedback/ParentFeedback.vue',
    'games/achievements.vue',
    'games/components/GameCard.vue',
    'games/index.vue',
    'games/play/AnimalObserver.vue',
    'games/play/ColorSorting.vue',
    'games/play/DinosaurMemory.vue',
    'games/play/DollhouseTidy.vue',
    'games/play/FruitSequence.vue',
    'games/play/PrincessGarden.vue',
    'games/play/PrincessMemory.vue',
    'games/play/RobotFactory.vue',
    'games/play/SpaceTreasure.vue',
    'games/records.vue',
    'profile/index.vue',
    'share-stats/index.vue'
]

print(f'🔧 开始修复 {len(pages_to_fix)} 个页面...\n')

fixed_count = 0
error_count = 0

for index, page in enumerate(pages_to_fix):
    file_path = os.path.join(parent_center_path, page)
    
    try:
        if not os.path.exists(file_path):
            print(f'⚠️ {index + 1}. {page} - 文件不存在')
            error_count += 1
            continue
        
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        if 'UnifiedCenterLayout' in content:
            print(f'✅ {index + 1}. {page} - 已使用UnifiedCenterLayout')
            fixed_count += 1
            continue
        
        # 替换硬编码的颜色值
        content = re.sub(r'#[0-9a-fA-F]{6}', 'var(--color-primary-500)', content, flags=re.IGNORECASE)
        content = re.sub(r'rgb\(\s*\d+\s*,\s*\d+\s*,\s*\d+\s*\)', 'var(--color-primary-500)', content)
        content = re.sub(r'rgba\(\s*\d+\s*,\s*\d+\s*,\s*\d+\s*,\s*[\d.]+\s*\)', 'var(--color-primary-500)', content)
        
        # 替换硬编码的px值
        def replace_px(match):
            px = int(match.group(1))
            if px <= 4:
                return ': var(--spacing-xs)'
            elif px <= 8:
                return ': var(--spacing-sm)'
            elif px <= 12:
                return ': var(--spacing-md)'
            elif px <= 16:
                return ': var(--spacing-lg)'
            elif px <= 20:
                return ': var(--spacing-xl)'
            elif px <= 24:
                return ': var(--spacing-2xl)'
            else:
                return match.group(0)
        
        content = re.sub(r':\s*(\d+)px', replace_px, content)
        
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f'✅ {index + 1}. {page} - 已修复')
            fixed_count += 1
        else:
            print(f'⚠️ {index + 1}. {page} - 无需修复')
    
    except Exception as e:
        print(f'❌ {index + 1}. {page} - 错误: {str(e)}')
        error_count += 1

print(f'\n📊 修复完成:')
print(f'  ✅ 成功: {fixed_count}')
print(f'  ❌ 失败: {error_count}')
print(f'  📝 总计: {len(pages_to_fix)}')

